Library/Homebrew/cmd/upgrade.rb

```diff
@@ -171,4 +177,35 @@ def upgrade_formula(f)
       nil
     end
   end
+
+  def relink_dependents(formula)
+    broken_formulae = []
+
+    CacheStoreDatabase.use(:linkage) do |db|
+      kegs = Formula.installed
+                    .collect(&:opt_or_installed_prefix_keg)
+                    .reject { |f| f.runtime_dependencies.to_a.empty? }
+
+      dependents = kegs.keep_if do |f|
+        f.runtime_dependencies
+         .any? { |d| d["full_name"] == formula.full_name }
+      end
+
+      break if dependents.empty?
+
+      ohai "Checking dependents of #{formula} for broken dynamic library links" if ARGV.verbose?
+
+      dependents.each do |keg|
+        checker = LinkageChecker.new(keg, cache_db: db)
+        broken_formulae << Formulary.from_keg(keg) if checker.broken_library_linkage?
+      end
+    end
+
+    return if broken_formulae.empty?
+
+    broken_formulae.each do |f|
```

Comment:
Should I deduplicate formulae here (e.g. in case one has multiple broken links)?

Reply:
@amyspark Yeh, I think it's worth doing 👍